lib/matplotlib/backends/backend_gtk4.py

```diff
@@ -0,0 +1,808 @@
+import functools
+import logging
+import os
+from pathlib import Path
+import sys
+
+import matplotlib as mpl
+from matplotlib import _api, backend_tools, cbook
+from matplotlib._pylab_helpers import Gcf
+from matplotlib.backend_bases import (
+    _Backend, FigureCanvasBase, FigureManagerBase, NavigationToolbar2,
+    TimerBase, ToolContainerBase)
+from matplotlib.backend_tools import Cursors
+from matplotlib.figure import Figure
+from matplotlib.widgets import SubplotTool
+
+try:
+    import gi
+except ImportError as err:
+    raise ImportError("The GTK4 backends require PyGObject") from err
+
+try:
+    # :raises ValueError: If module/version is already loaded, already
+    # required, or unavailable.
+    gi.require_version("Gtk", "4.0")
+except ValueError as e:
+    # in this case we want to re-raise as ImportError so the
+    # auto-backend selection logic correctly skips.
+    raise ImportError from e
+
+from gi.repository import Gio, GLib, GObject, Gtk, Gdk
+
+
+_log = logging.getLogger(__name__)
+
+backend_version = "%s.%s.%s" % (
+    Gtk.get_major_version(), Gtk.get_micro_version(), Gtk.get_minor_version())
```

Comment:
looks like minor/micro are switched...

Reply:
Ah looks like this has been wrong in the GTK3 backend too, since it was introduced.